cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -369,30 +371,53 @@ func ValidateHostPort(endpoint string, fldPath *field.Path) field.ErrorList {
 }
 
 // ValidateIPNetFromString validates network portion of ip address
-func ValidateIPNetFromString(subnet string, minAddrs int64, fldPath *field.Path) field.ErrorList {
+func ValidateIPNetFromString(subnetStr string, minAddrs int64, checkDualStack bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	_, svcSubnet, err := net.ParseCIDR(subnet)
-	if err != nil {
-		allErrs = append(allErrs, field.Invalid(fldPath, subnet, "couldn't parse subnet"))
-		return allErrs
-	}
-	numAddresses := ipallocator.RangeSize(svcSubnet)
-	if numAddresses < minAddrs {
-		allErrs = append(allErrs, field.Invalid(fldPath, subnet, "subnet is too small"))
+	if checkDualStack {
+		subnets, err := utilnet.ParseCIDRs(strings.Split(subnetStr, ","))
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, subnetStr, err.Error()))
+		} else {
+			isDualStack, err := utilnet.IsDualStackCIDRs(subnets)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath, subnets, err.Error()))
+			} else if !isDualStack {
```

Comment:
would this mean that if the user enables the feature gate, but the list of CIDRs they pass does not match both stacks https://github.com/kubernetes/utils/blob/c55fbcfc754a5b2ec2fbae8fb9dcac36bdba6a12/net/net.go#L80-L81 that the validation will fail.

my understanding of the feature is the following, but i could be wrong :)
- if the users passes multiple CIDRs
  - if they are from different families and if the feature gate is not enabled return error
  - if they are from the same family the feature gate is a NO-OP and we don't error out
- if the user passes a single CIDR, the feature gate is a NO-OP

please correct me, as i could be lacking some context.

Reply:
> * if they are from the same family the feature gate is a NO-OP and we don't error out

This is not correct. Multiple CIDRs are only supported when FG is enabled and the must contain at least one CIDR from each family.

Reply:
ok, i see.